### PR TITLE
Add description of `namespaces.yaml`

### PR DESCRIPTION
**Pull request by fbertsch**
I wanted to put this in there for us to start thinking more clearly about the relationship between Views, Explores, and `namespaces.yaml`. I understand it's confusing and want to make the architecture more concrete.

### CODE REVIEW

File: architecture/namespaces_yaml.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+We use [`namespaces.yaml`](https://github.com/mozilla/looker-hub/blob/main/namespaces.yaml) as the declarative listing of what we will generate.
+
+Each namespace in `namespaces.yaml` has the following:
+
+`owners` (string): The owners are the people who will have control over the associated Namespace folder in Looker. It is up to them to decide which dashboards to "promote" to their shared folder.
+`pretty_name` (string): The pretty name is used in most places where the namespace's name is seen, e.g. in the explore drop-down and folder name.
+`glean_app` (bool): Whether or not this namespace represents a Glean Application.
+`connection` (optional string): The database connection to use, as named in Looker. Defaults to `telemetry`.
+`views`: The LookML View files that will be generated. More detailed info below.
```
Comment (acmiyaguchi) [inline architecture/namespaces_yaml.md:11]: (object)

File: architecture/namespaces_yaml.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+## `views`
+
+Each view is actuall a LookML view file that will be generated.
+Each LookML View file can contain multiple Looker Views; the idea here is that these views are related and used together.
+
+`type` - The type of the view, e.g. `glean_ping_view`.
+`tables` - This field is used in a few ways, depending on the associated View type.
+
+For `GleanPingView` and `PingView`, `tables` represents all of the associated channels for that view. Each table will have a `channel` and `table` entry. Only a single view will be created in the LookML File.
+```
```
Comment (acmiyaguchi) [inline architecture/namespaces_yaml.md:23]: Add the file type to the code fences.

```suggestion
```yaml
```

File: architecture/namespaces_yaml.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Describing the formatting of `namespaces.yaml`
+
+We use [`namespaces.yaml`](https://github.com/mozilla/looker-hub/blob/main/namespaces.yaml) as the declarative listing of what we will generate.
```
Comment (wlach) [inline architecture/namespaces_yaml.md:3]:
```suggestion
We use [`namespaces.yaml`](https://github.com/mozilla/looker-hub/blob/main/namespaces.yaml) as the declarative listing of the Looker namespaces generated by this repository.
```

File: architecture/namespaces_yaml.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Each namespace in `namespaces.yaml` has the following:
```
Comment (wlach) [inline architecture/namespaces_yaml.md:5]:
```suggestion
Each entry in `namespaces.yaml` represents a namespace, and has the following properties:
```

File: architecture/namespaces_yaml.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+`views`: The LookML View files that will be generated. More detailed info below.
+`explores`: The LookML Explore files that will be generated. More detailed info below.
+
+## `views`
+
+Each view is actuall a LookML view file that will be generated.
+Each LookML View file can contain multiple Looker Views; the idea here is that these views are related and used together.
+
+`type` - The type of the view, e.g. `glean_ping_view`.
+`tables` - This field is used in a few ways, depending on the associated View type.
```
Comment (wlach) [inline architecture/namespaces_yaml.md:20]:
```suggestion
`type`: The type of the view, e.g. `glean_ping_view`.
`tables`: This field is used in a few ways, depending on the associated View type.
```

File: architecture/namespaces_yaml.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+tables:
+  - funnel_analysis: events_daily_table
+    event_types: `mozdata.glean_app.event_types`
+    event_type_1: event_types
+    event_type_2: event_types
+```
+
+## `explores`
+
+Each Explore is a single Explore file, sometimes containing multiple explores within it (mainly for things like changing suggestions).
```
Comment (acmiyaguchi) [inline architecture/namespaces_yaml.md:49]: "Each Explore entry"

**Review by acmiyaguchi — APPROVED** [review]
General description of the namespace yaml looks good to me.

File: architecture/namespaces_yaml.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+For `ClientCountView` and `GrowthAccountingView`, `tables` will have a single entry, with the name of the table the Looker View is based off of. Only a single Looker View will be created.
+```
+tables:
+  - table: mozdata.org_mozilla_firefox.baseline_clients_last_seen
+```
+
+For `FunnelAnalysisView`, only the first list entry is used; inside that entry, each value represents a Looker View that is created. The key is the name of the view, the value is the Looker View or BQ View it is derived from.
+In the following example, 4 views will be created in the view file.
```
Comment (wlach) [inline architecture/namespaces_yaml.md:38]:
```suggestion
In the following example, 4 views will be created in the view file: `funnel_analysis`, `event_types`, `event_type_1` and `event_type_2`. 
```

File: architecture/namespaces_yaml.md (hunk 6 above)
Comment (wlach) [inline architecture/namespaces_yaml.md:49]:
```suggestion
Each Explore is a single file, sometimes containing multiple explores within it (mainly for things like changing suggestions).
```

File: architecture/namespaces_yaml.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# Describing the formatting of `namespaces.yaml`
```
Comment (wlach) [inline architecture/namespaces_yaml.md:1]: I think just namespaces.yaml makes a fine title

```suggestion
# `namespaces.yaml`
```

**Review by wlach — APPROVED** [review]
Thanks for writing this! In addition to the nits, writing something about how namespaces relate to Glean applications could be helpful in the future. r+, but feel free to ask for more review.

**Comment by fbertsch** [comment]
Thank you @wlach and @acmiyaguchi! We'll continue to iterate on these docs, I imagine some of this belongs in the classes themselves and other parts should be clarified by improving the namespaces.yaml structure. (And more concretely, the relationship to Glean.)

**Pull request merged**